src/xtgeo/grid3d/grid.py

```diff
@@ -1310,6 +1312,42 @@ def get_dz(self, name="dZ", flip=True, asmasked=True, mask=None):
 
         return deltaz
 
+    def get_dx(self, name="dX", asmasked=True):
+        """Return the dX and dY as GridProperty object.
+
+        The values lengths are projected to a constant Z.
+
+        Args:
```

Comment:
Will it be simple to get variant not projected to constant Z? E.g. buy adding a boolean key `zprojection`

Reply:
It is possible, but is there a use case? One thing I find a bit troubling is that `get_dx` and `get_dz` have different definitions, `get_dx` projects onto constant Z, while `get_dz` is not the edge length but the height difference. Perhaps we should create an issue for deciding one definition to rule them all? In that case, just an average edge length seems the most logical.

Reply:
The use case is that RMS uses both: `x_increment` (projected to same Z) and `x_size` (total distance), so they can be applied for various use cases. Do you use all 4 corners in you code? (get_dx should be the avg. lengths of upper and lower edges. For me `get_dz()` is the average vertical thickess, projected to the normal of the Z plane

Reply:
Is there a RMS equivalent for `get_dz` ?

Reply:
The code uses all 4 corners.

Reply:
> Is there a RMS equivalent for `get_dz` ?

Yes there are; think it just computes vertical thickness (i.e. along the normal vector to the Z plane)